### PR TITLE
fallback to Nav Native's internal TileStore creation (with correct access token and path) if one is not provided

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Fallback to Nav Native's internal `TileStore` creation (with correct access token and path) if one is not provided. Before this change, if `TileStore` was not provided, we would fallback to creating an empty `TileStore` with no token, which means that Common SDK would use the `mapbox_access_token` string resource to obtain the token. If that resource was not available, the token would be empty, and routing tiles requests would fail.

### Changelog

```
<changelog>Fixed an issue where routing tiles were not downloading correctly if the `TileStore` instance was not provided and `mapbox_access_token` string resource was not present. This was resulting in a lack of free drive events and location enhancement failures.</changelog>
```

### CODE REVIEW

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt [hunk 1]
```diff
@@ -22,7 +22,9 @@ import java.net.URI
  * @param filePath used for persistent configuration and history files storing.
  * @param tileStore tile store instance. It manages downloads and storage for requests to
  * tile-related API endpoints. For offline/predictive-caching use cases this instance should be
- * the same that is passed to map resource options.
+ * the same that is passed to map resource options. When creating the [TileStore] make sure to call
+ * [TileStore.setOption] with [TileStoreOptions.MAPBOX_ACCESS_TOKEN] and your token.
+ * By default (if `null` is provided here), the [TileStore] will be created with a [filePath] and [NavigationOptions.accessToken].
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt:27]: @mskurydin could you double-check that this is all true?

Comment (korshaknn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt:27]: should we mention domain?
`TileDataDomain.NAVIGATION`

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt [hunk 2]
```diff
@@ -177,10 +179,12 @@ class RoutingTilesOptions private constructor(
 
         /**
          * Override tile store instance. It manages downloads and storage for requests to
-         * tile-related API endpoints. For offline/predictive-caching use cases this instance
-         * should be the same that is passed to map resource options.
+         * tile-related API endpoints. For offline/predictive-caching use cases this instance should be
+         * the same that is passed to map resource options. When creating the [TileStore] make sure to call
+         * [TileStore.setOption] with [TileStoreOptions.MAPBOX_ACCESS_TOKEN] and your token.
+         * By default (if `null` is provided here), the [TileStore] will be created with a [filePath] and [NavigationOptions.accessToken].
```
Comment (korshaknn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt:185]: the same. should we mention `TileDataDomain.NAVIGATION` ?

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/options/RoutingTilesOptionsTest.kt [hunk 3]
```diff
@@ -93,19 +75,9 @@ class RoutingTilesOptionsTest : BuilderTest<RoutingTilesOptions, RoutingTilesOpt
...
-
-    @Test
-    fun `custom TileStore instance overrides a default one`() {
-        val customTileStore = TileStoreProvider.getTileStoreInstance("custom_tile_store_path")
-        val options = RoutingTilesOptions.Builder()
-            .tileStore(customTileStore)
-            .build()
-
-        assertEquals(customTileStore, options.tileStore)
+        Assert.assertNull(defaultOptions.tileStore)
```
Comment (Guardiola31337) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/options/RoutingTilesOptionsTest.kt:81]: NIT

```suggestion
        assertNull(defaultOptions.tileStore)
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt:27]: I don't think we have to. It should be very a rare occasion where a developer needs to specify a domain, normally, they can use the same token for all of their usages.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt (hunk 2 above)
Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/options/RoutingTilesOptions.kt:185]: Same as above.

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**